packages/core/src/editor/NodeHelpers.ts

```diff
@@ -0,0 +1,189 @@
+import { EditorState, Node, NodeId } from "@craftjs/core";
+import invariant from "tiny-invariant";
+import {
+  deprecationWarning,
+  ERROR_CANNOT_DRAG,
+  ERROR_DUPLICATE_NODEID,
+  ERROR_INVALID_NODE_ID,
+  ERROR_MOVE_INCOMING_PARENT,
+  ERROR_MOVE_NONCANVAS_CHILD,
+  ERROR_MOVE_OUTGOING_PARENT,
+  ERROR_MOVE_TO_DESCENDANT,
+  ERROR_MOVE_TO_NONCANVAS_PARENT,
+  ERROR_MOVE_TOP_LEVEL_NODE,
+  ROOT_NODE,
+} from "@craftjs/utils";
+import { serializeNode } from "../utils/serializeNode";
+import { mergeTrees } from "../utils/mergeTrees";
+
+export function NodeHelpers(state: EditorState, id: NodeId) {
+  invariant(typeof id == "string", ERROR_INVALID_NODE_ID);
+
+  const node = state.nodes[id];
+
+  const ref = (id) => NodeHelpers(state, id);
+
+  const getNodeFromIdOrNode = (node: NodeId | Node) =>
+    typeof node === "string" ? state.nodes[node] : node;
+
+  return {
+    isCanvas() {
+      return !!node.data.isCanvas;
```

Comment:
btw at some ponint we should rething the `data` and also the `isCanvas` property.

Reply:
So I remembered another reason why I kept the `type` prop in `data` - it's because when we're serialising, it's just a lot easier to to just get everything form `data`. About `isCanvas`, I'm thinking if we should rename it to just `canvas` to match `<Element />`'s canvas prop